android/app/src/main/java/com/sample/react/TestAppReactNativeHost.kt

```diff
@@ -1,22 +1,47 @@
 package com.sample.react
 
 import android.app.Application
-import com.facebook.react.PackageList
-import com.facebook.react.ReactNativeHost
-import com.facebook.react.ReactPackage
+import com.facebook.react.*
+import com.facebook.react.bridge.ReactMarker
+import com.facebook.react.bridge.ReactMarkerConstants
+import com.facebook.react.common.LifecycleState
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class TestAppReactNativeHost @Inject constructor(
-    application: Application,
-    private val reactBundleNameProvider: ReactBundleNameProvider
+        application: Application,
+        private val reactBundleNameProvider: ReactBundleNameProvider
 ) : ReactNativeHost(application) {
 
-    override fun getBundleAssetName() =
-        reactBundleNameProvider.bundleName ?: super.getBundleAssetName()
+    var currentActivity: ReactActivity? = null
```

Comment:
As said earlier, I don't think we need to keep track of currentActivity `ourselves`.